File: wdio-ios-app/tests/pages/menu.page.ts
import { $ } from '@wdio/globals';

class MenuPage {
  get loginButton() {
    return $('~LogOut-menu-item');
  }

  get menuButton() {
    return $('~More-tab-item');
  }

  /**
   * Open the menu
   */
  async open() {
    await this.menuButton.click();
  }

  /**
   * Click Login Menu Button
   */
  async clickLoginButton() {
    await this.loginButton.click();
  }

  async waitForIsShown(selector: any): Promise<boolean> {
    const result = await selector.waitForDisplayed({
      timeout: 30 * 1000,
    });
    return !!result;
  }
}

export default new MenuPage();

File: java-ios-app/src/test/java/com/example/pageobjects/MenuPage.java
package com.example.pageobjects;

import io.appium.java_client.AppiumBy;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebDriver;

public class MenuPage {

  private final RemoteWebDriver driver;

  public MenuPage(RemoteWebDriver driver) {
    this.driver = driver;
  }

  public WebElement getLoginButton() {
    return driver.findElement(AppiumBy.accessibilityId("LogOut-menu-item"));
  }

  public WebElement getMenuButton() {
    return driver.findElement(AppiumBy.accessibilityId("More-tab-item"));
  }

  public void open() {
    getMenuButton().click();
  }

  public void clickLoginButton() {
    getLoginButton().click();
  }
}
